scripts/ci/ensure-npm-shrinkwrap-versions.sh

```diff
@@ -26,15 +26,15 @@ HERE="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 SHRINKWRAP_JSON=npm-shrinkwrap.json
 
 # Two pair-wise arrays, because associative arrays only work in Bash 4
-MODULE_NAMES=("dependencies[\"node-gyp\"]")
-MODULE_VERSIONS=("3.5.0")
+MODULE_NAMES=()
+MODULE_VERSIONS=()
 
 if [[ ${#MODULE_NAMES[@]} -ne ${#MODULE_VERSIONS[@]} ]]; then
     echo "Lengths of MODULE_NAMES and MODULE_VERSIONS arrays must match"
     exit 1
 fi
 
-for i in ${!MODULE_NAMES[@]}; do
+for i in "${!MODULE_NAMES[@]}"; do
```

Comment:
Are you sure the change on this line is necessary? I've just been doing some local testing, and regardless of whether there's 0, 1 or 2 entries in the `MODULE_NAMES` array, this line works just fine without the quotes.

Reply:
That's just a mend from a [shellcheck](https://github.com/koalaman/shellcheck) warning to "avoid array expansions and re-splitting elements [SC2068]".

Reply:
Side-note: Would it be worth adding shellcheck to our lint-step?

Reply:
Generally I'd like to, but installing it on Windows might be a bit of a pain, plus sorting out the false positives via config – we're probably better off occasionally running it ourselves.

I think the general solution here is to just not have bash scripts in the first place – ideally we'd port them to JS, as we have a cross-platform runtime w/ Node anyways, and we wouldn't have to worry about diff bash versions, and could also eliminate it as a dep on Windows.

Reply:
Ahh, I thought that the reason that these scripts were written in Bash was so that we had an "out of band" mechanism, so that the scripts would still run if something went wonky with the nodeJS setup? (But perhaps that was a false assumption on my part? I agree that these would be much easier to maintain if written in a "proper" programming language, rather than trying to squeeze them into Bash's limitations)

Reply:
> so that the scripts would still run if something went wonky with the nodeJS setup?

If something with that was to go wrong, then running anything else wouldn't make much sense imo (or am I missing something?) – so we might as well dump bash.

Reply:
Ping @jviotti - is this something worth attempting? (after we've got the next release out of the way, obviously)

Reply:
Doing it gradually would be a relaxed approach, doing it all at once might be a bit heavy and bug-prone